cmd/mt-fakemetrics/cmd/datafeed.go

```diff
@@ -211,9 +225,30 @@ times %4d orgs: each %s, flushing %d metrics so rate of %d Hz. (%d total unique
 				// we already sent, so we must increase the timestamp
 				if m == 0 {
 					ts += mp
+					if vp.Policy == policy.Multiple {
+						if (idxValues + 1) >= lenValues {
+							idxValues = 0
+						} else {
+							idxValues++
+						}
+					}
 				}
 				metricData.Time = ts
-				metricData.Value = rand.Float64() * float64(m+1)
+
+				switch vp.Policy {
+				case policy.None:
```

Comment:
I think instead of switch/case and if conditions in this function which change the behavior based on the specified policy, it would be nicer to make a

```
type Policy interface {
  GetValue(ts int64) float64
}
```

and then just create different types for each policy which satisfy this interface. that would make `dataFeed()` cleaner

Reply:
I thought about doing that initially. I don't have any issues with it.